atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -121,15 +148,68 @@ public void close() {
                     SafeArg.of("poolRefreshIntervalSeconds", config.poolRefreshIntervalSeconds()),
                     e);
 
-            // return the set of servers we knew about last time we successfully constructed the tokenMap
-            return tokenMap.asMapOfRanges().values().stream().flatMap(Collection::stream).collect(Collectors.toSet());
+            // Attempt to re-resolve addresses from the configuration; this is important owing to certain race
+            // conditions where the entire pool becomes invalid between refreshes.
```

Comment:
you've somehow managed to make other changes than your PR contained. Are you sure you merged changes in correctly?